### PR TITLE
SYNPY-582

### PR DESCRIPTION
**Pull request by kimyen**

### CODE REVIEW

File: synapseclient/client.py [hunk 1]
```diff
@@ -675,6 +675,28 @@ def _getFromFile(self, filepath, limitSearch=None):
 
         return bundle
 
+    def move(self, entity, new_parent):
+        """
+        Move a Synapse entity to a new container.
+
+        :param entity:           A Synapse ID, a Synapse Entity object, or
+                                 a local file that is stored in Synapse (found by hash of file)
```
Comment (kimyen) [inline synapseclient/client.py]: remove "(found by hash of file)"

File: synapseclient/client.py [hunk 2]
```diff
@@ -675,6 +675,28 @@ def _getFromFile(self, filepath, limitSearch=None):
...
+        :param new_parent:       The new parent container (Folder or Project)
+                                 to which the entity should be moved.
+
+        :returns: The Synapse Entity object that has been moved.
+
+        Example::
+
+            entity = syn.move('syn456', 'syn123')
```
Comment (kimyen) [inline synapseclient/client.py]: what happen when user try to move a project?

File: tests/integration/integration_test.py [hunk 3]
```diff
@@ -522,3 +522,7 @@ def testSetStorageLocation__existing_storage_location():
     new_retrieved_setting = syn.getProjectSetting(proj, 'upload')
     assert_equals(new_storage_setting, new_retrieved_setting)
 
+def testMoveProject():
+    proj1 = syn.store(Project(name=str(uuid.uuid4()) + "testMoveProject-child"))
+    proj2 = syn.store(Project(name=str(uuid.uuid4()) + "testMoveProject-newParent"))
```
Comment (zimingd) [inline tests/integration/integration_test.py]: use `schedule_for_cleanup(proj)` to clean up created entities in integration tests

File: tests/unit/unit_test_client.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 from nose.tools import assert_equal, assert_in, assert_raises, assert_is_none
 
 import synapseclient
-from synapseclient import Evaluation, File, Folder
+from synapseclient import Evaluation, File, Folder, Project
```
Comment (zimingd) [inline tests/unit/unit_test_client.py]: remove import if unused

**Pull request merged**